CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+
+All of these are automatically built and deployed by our [Jenkins build servers](https://builds.renegadetech.mixxx.org/) whenever a change is committed. Anyone is welcome to open a pull request for all of these repositories.
+
+## Git Workflow ##
+### All Contributors ###
+* Each feature/bug fix should be done on its own Git branch so they can be reviewed and merged independently. Refer to [Using Git](https://mixxx.org/wiki/doku.php/using_git) for how to do this. Please ask for help on [Zulip](https://mixxx.zulipchat.com/) if you have questions about using Git after reading that page.
+* Commits should be as small as they can while still building. The smaller the commit, the easier it is to review. It also makes it easier to revert if it is later identified as the source of a bug. If you have lots of changes that you need to commit, a [GUI Git client](https://git-scm.com/downloads/guis) can be helpful for picking out specific changes for multiple small commits.
+* Every commit should build. This is important so [git bisect](https://git-scm.com/book/en/v2/Git-Tools-Debugging-with-Git#_binary_search) works.
+* Commit messages should succinctly describe what is changed in that commit and why. Lines should wrap at 72 characters so they show fully in GitHub and other Git tools. For example, this is a good commit message:
+
+    ```
+    DlgPrefEffects: add QListWidget to set order of chains
+
+    This order will soon be used by new ControlObjects to load them
+    from controllers.
+    ```
+
+    This is not a good commit message:
+
+    ```
+    address comments from PR review
+    ```
+
+    Neither is this:
+
+    ```
+    fix a bug with quantize while the deck is playing and master sync is enabled and an effect unit is on the deck while the user is turning an EQ knob
+    ```
+
+    Refer to [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/) for more details.
+
+* Install [pre-commit](https://pre-commit.com/#install) to automatically ensure that your commits comply with our code style for both C++ and JavaScript. Note this is currently not working on Windows. This saves time reviewing so we don't have to point out nitpicky style issues. Once you have pre-commit installed on your computer, set it up in your local Git repository:
+
+    ```
+    cd /path/to/your/git/repo
+    pre-commit install
+    pre-commit install -t pre-push
+    ```
+
+    If you have a problems with a particular hook, you can use the SKIP environment variable to disable hooks:
+
+    ```
+    SKIP=clang-format,end-of-file-fixer git commit
+    ```
+
+    This can also be used to separate logic changes and autoformatting into two subsequent commits. Using the SKIP environment variable is preferable to using `git commit --no-verify` (which also disables the checks) because it won't prevent catching other, unrelated issues.
```

Comment:
Please divide this into sections, it is rather unwieldy if you are searching for something specific.

E.g. this section could be called "Committing"